google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerOptions.java

```diff
@@ -1205,6 +1216,7 @@ public SpannerOptions build() {
             this.grpcGcpExtensionEnabled ? GRPC_GCP_ENABLED_DEFAULT_CHANNELS : DEFAULT_CHANNELS;
       }
 
+      SpannerRpcMetrics.initializeRPCMetrics(this.openTelemetry);
```

Comment:
I don't think we want this here (although I don't really know what it does). But it seems strange to initialize some metrics as a side-effect of building an instance of `SpannerOptions`. At this time, we have still not opened a connection to `Spanner`. I would expect this rather to happen when you call `SpannerOptions#getService()` (it does not need to be in that method, but maybe in the constructor of any of the objects that are being created by that call)

Reply:
This will initialise gRPC metrics `gfe_latency` and `gfe_header_count`. These are gRPC metrics which will be applicable for every gRPC call and data for this is added in `headerIntercepetor` class. 

I was looking a way to initialise these metrics before any gRPC call can happen.  Customers will use SpannerOptions to initialise the `OpenTelemetry` object, once the object is passed we can initialise these metrics. 

What would be the best place to do that ? Should it be in `SpannerOptions#getService()` ? Are we sure this method will be called with all the possible ways customer might use to initialise spanner object ?

Reply:
`SpannerOptions#getService()` is always called before a user can call any gRPC methods. But I think that in this case  it would be better to put this call in the `SpannerImpl` constructor.

Reply:
done